projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -126,4 +154,26 @@ export class CodeViewerComponent implements OnChanges {
   public onSearch(searchText: string): void {
     this.searchText = searchText;
   }
+
+  private observeDomMutations(): void {
+    const codeViewerContentElement = this.element.nativeElement.querySelector('#code-viewer-content') as Node;
+    this.domMutationObserver.observe(codeViewerContentElement, {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      attributeFilter: ['class']
+    });
+  }
+
+  private onDomMutation(mutations: MutationRecord[]): void {
```

Comment:
If a text search creates 10 mark elements, would we call this method 10 times?

Reply:
Can we do this instead? 
```
  public readonly debouncedOnDomMutation: (mutations: MutationRecord[]) => void = debounce(
    this.onDomMutation,
    500,
     {leading: true}
  );
  
  ```

Reply:
Can we do this instead? 
```
  public readonly debouncedOnDomMutation: (mutations: MutationRecord[]) => void = debounce(
    this.onDomMutation,
    500,
     {leading: true}
  );
  
  ```

Reply:
We can do it, but for search debounce has already been added in the `search-box` component, do we really need this?
To this we would be making the search box debounce as `0` and then I guess we can surely use this

Reply:
This debounce would be for avoiding multiple multiple mutation action. Since, you mentioned this is not happening in the other comment then i guess we don't need it.